src/objective-c/GRPCClient/private/GRPCChannel.m

```diff
@@ -33,22 +33,101 @@
 
 #import "GRPCChannel.h"
 
-#include <grpc/grpc.h>
+#include <grpc/grpc_security.h>
+#include <grpc/support/alloc.h>
+#include <grpc/support/log.h>
 
-@implementation GRPCChannel
+/**
+ * Returns @c grpc_channel_credentials from the specifie @c path. If the file at the path could not
+ * be read then NULL is returned.  If NULL is returned, @c errorPtr may not be NULL if there are
+ * details available describing what went wrong.
+ */
+static grpc_channel_credentials *CertificatesAtPath(NSString *path, NSError **errorPtr) {
+  // Files in PEM format can have non-ASCII characters in their comments (e.g. for the name of the
+  // issuer). Load them as UTF8 and produce an ASCII equivalent.
+  NSString *contentInUTF8 = [NSString stringWithContentsOfFile:path
+                                                      encoding:NSUTF8StringEncoding
+                                                         error:errorPtr];
+  NSData *contentInASCII = [contentInUTF8 dataUsingEncoding:NSASCIIStringEncoding
+                                       allowLossyConversion:YES];
+  if (!contentInASCII.bytes) {
+    // Passing NULL to grpc_ssl_credentials_create produces behavior we don't want, so return.
+    return NULL;
+  }
+  return grpc_ssl_credentials_create(contentInASCII.bytes, NULL, NULL);
+}
+
+/**
+ * Allocates a @c grpc_channel_args and populates it with the options specified in the
+ * @c dictionary. Keys must be @c NSString.  If the value responds to @c @selector(UTF8String) then
+ * it will be mapped to @c GRPC_ARG_STRING.  If not, it will be mapped to @c GRPC_ARG_INTEGER if the
+ * value responds to @c @selector(intValue).  Otherwise, an exception will be raised.
+ */
+grpc_channel_args * buildChannelArgs(NSDictionary *dictionary) {
+  if (!dictionary) {
+    return NULL;
+  }
+
+  NSUInteger argCount = [dictionary count];
+
+  // Allocate memory for both the individual args and their container
+  grpc_channel_args *channelArgs = gpr_malloc(sizeof(grpc_channel_args)
```

Comment:
Is it expected that C grpc will free it with `gpr_free`, or is it up to the caller of this function?